common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -44,6 +49,9 @@
   public static final int DOUBLE_ARRAY_OFFSET;
 
   private static final boolean unaligned;
+
+  private static final MemoryBlock _doubleBuffer;
```

Comment:
Admittedly, I am not deeply familiar with how this class is being using concurrently within a single JVM. If multiple threads are using it concurrently, you are right and a different approach must be found. I was trying to avoid the cost of reallocating a buffer with ever call to `getDouble()`.